Update - August 2020 - New filters on product structure API endpoints (#739)
akeneo/pim-helpcenter: content/updates/2020-08/2-API-search.md

---
pim_announcement_img: ./img/api-filters.png
pim_announcement_alt_img: "API documentation about attribute filters"
pim_announcement_audience:
- CE
- EE
---

# New updates to ease attribute synchronization with the API

> ::: meta-data type="New" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/documentation/filter.html"

Reviewer: ```suggestion
::: meta-data type="New" features="API, Connectivity" available="early August" link-to-doc="https://api.akeneo.com/documentation/filter.html"
```

---
pim_announcement_img: ./img/api-filter-attributes.png
pim_announcement_alt_img: "API documentation about attribute filters"
pim_announcement_audience:
- CE
- EE
---

> # New filters on product structure API endpoints

Reviewer 2: ```suggestion
# Search through API calls improved
```
not totally satisfied with this suggestion.. what do you think?

Author: Only get what you need from the API?

::: meta-data type="New" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/documentation/filter.html"
:::

> With the new `search` parameter, you can now add filters on product structure endpoints: attributes, attribute groups, categories, and families.

Reviewer 2: ```suggestion
With the new `search` parameter, you can now add filters on product structure endpoints: attributes, attribute groups, categories, and families. This new parameter will help reducing the number of API calls needed to synchronize product data. 
```

Author: This is not exactly the role of these filters. But it will reduce the size of the API call responses and the number of data to process.

> With the new `search` parameter, you can now add filters on product structure endpoints: attributes, attribute groups, categories, and families.

Reviewer: ```suggestion
With the new `search` parameter, you can now add filters on product structure endpoints: attributes, attribute groups, categories, and families.  
```

For example, you can retrieve data for all attributes that have been updated since a specific date, or you can get the data from all the sub-categories of a specific tree.

::: more

> [How to filter attributes?](https://api.akeneo.com/documentation/filter.html#filter-attributes)

Reviewer: ```suggestion
[How to filter attributes?](https://api.akeneo.com/documentation/filter.html#filter-attributes)  
```

> [How to filter attribute groups?](https://api.akeneo.com/documentation/filter.html#filter-attribute-groups)

Reviewer: ```suggestion
[How to filter attribute groups?](https://api.akeneo.com/documentation/filter.html#filter-attribute-groups)  
```

> [How to filter families?](https://api.akeneo.com/documentation/filter.html#filter-families)

Reviewer: ```suggestion
[How to filter families?](https://api.akeneo.com/documentation/filter.html#filter-families)  
```

---
pim_announcement_img: ./img/api-filter-attributes.png
pim_announcement_alt_img: "API documentation about attribute filters"
pim_announcement_audience:
- CE
- EE
---

# New filters on product structure API endpoints
::: meta-data type="New" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/documentation/filter.html"
:::

With the new `search` parameter, you can now add filters on product structure endpoints: attributes, attribute groups, categories, and families.
For example, you can retrieve data for all attributes that have been updated since a specific date, or you can get the data from all the sub-categories of a specific tree.

::: more
[How to filter attributes?](https://api.akeneo.com/documentation/filter.html#filter-attributes)
[How to filter attribute groups?](https://api.akeneo.com/documentation/filter.html#filter-attribute-groups)
[How to filter families?](https://api.akeneo.com/documentation/filter.html#filter-families)
[How to filter categories?](https://api.akeneo.com/documentation/filter.html#filter-categories)
:::

> # Option labels are now available in the product endpoint response

Reviewer: It should be in a separate file. One feature, one file. Otherwise, it won't appear in the PIM.
If the two features are too small to be advertised separately, you can group them in the same file, but you'll need to have only one level-one title with inside two subsections with level-two titles. Don't hesitate to ping me if not clear.

---
pim_announcement_img: ./img/api-filters.png
pim_announcement_alt_img: "API documentation about attribute filters"
pim_announcement_audience:
- CE
- EE
---

# Only get what you need thanks to our new API filter
::: meta-data type="New" features="API, Connectivity" available="late August" link-to-doc="https://api.akeneo.com/documentation/filter.html"
:::

> We added a new `search` parameter to reduce the number of data to process and the size of the API call responses.

Reviewer: ```suggestion
We added a new `search` parameter to reduce the number of data to process and the size of the API call responses.  
```
...

> You can now add filters on product structure endpoints. 

Reviewer: ```suggestion
You can now add filters on catalog structure endpoints. 
```
And which endpoints?

Here is the list of the available filters:

> - updated date, 

Reviewer: What's the use case?

Reviewer: And fo which endpoint is it available?

> - code, 

Reviewer: What's the use case?

Reviewer: And for which endpoint is it available?

> - type: only available for attributes. It allows you to filter all attributes of a specific type. 

Reviewer: What's the use case?

> - parent: only available for categories. Thanks to this filter, you can retrieve all sub-categories of a parent category. 

Reviewer: What's the use case?

> For example, the following filter will help you to retrieve all the simple select attributes that are mentionned in the `code` list, and have been updated since the 2020-07-23 at 15:19.

Reviewer: Do we have a use case we could describe here that would explain why this combination of filters is really powerful?

---
pim_announcement_img: ./img/api-filters.png
pim_announcement_alt_img: "API documentation about attribute filters"
pim_announcement_audience:
- CE
- EE
---

# New updates to ease API catalog structure synchronization 
::: meta-data type="New" features="API, Connectivity" available="early August" link-to-doc="https://api.akeneo.com/documentation/filter.html"
:::

A **new `search` parameter** enables you to add filters when retrieving attributes or families.

> Those filters will help reduce the number of data to process and the size of the API call responses. 

Reviewer 3: It reduces the number of API calls needed to synchronize systems easing delta synchronization on a list of products.
I don't know if it's more explicit like that :D

---
pim_announcement_img: ./img/api-filters.png

> pim_announcement_alt_img: "API documentation about attribute filters"

Reviewer 2: @NolwennP do we need those if we don't have images in the PR?

pim_announcement_audience:
- CE
- EE
---

> # Only get what you need thanks to our new API filter

Reviewer: ```suggestion
# Only get what you need thanks to our new API filters
```

Reviewer: I would suggest something like this:
```suggestion
# Efficient catalog structure export thanks to new filters
```

> # Only get what you need thanks to our new API filter

Reviewer: To be straight to the point, I would suggest:
```suggestion
# Efficient catalog structure export via API thanks to new filters
```

Author: ```suggestion
# New filters to simplify API catalog structure exports
```

::: meta-data type="New" features="API, Connectivity" available="early August" link-to-doc="https://api.akeneo.com/documentation/filter.html"
:::

> A **new `search` parameter** enables you to add filters when retrieving attributes or families. Moreover, we added **attribute group labels** inside the attribute endpoints.

Reviewer: ```suggestion
A **new `search` parameter** enables you to add filters when retrieving attributes or families.
```

> Those two enhancements will help reduce the number of data to process, the size of the API call responses, and the number of API calls needed to synchronize your catalog structure. 

Reviewer: ```suggestion
Those filters will help reduce the number of data to process and the size of the API call responses. 
```

- **updated date**: allows you to retrieve only data that have been updated since a specific date and hour (available for attribute groups, attributes, and families), 
- **code**: allows you to filter on specific codes (available for attribute groups, attributes, categories and families), 
- **type**: allows you to filter on attributes from a specific type (only available for attributes),
- **parent**: retrieves all sub-categories of a parent category (only available for categories).  

You can combine filters to achieve powerful results!

Let's take an example: you have an e-commerce website that uses your PIM data. You know the list of all the attributes you need, and your application synchronises the catalog structure every day at 3PM.  

> The following filter will help you retrieve all the attributes mentioned in the `code` list that have been updated since the 2020-08-26 at 15:00:

Author: It's not like with databases, it's not a query even if it looks like. It's a URL parameter actually

> The following filter will help you retrieve all the attributes mentioned in the `code` list that have been updated since the 2020-08-26 at 15:00:

Author: the following example maybe ?

```

> /api/rest/v1/attributes?search={"updated": [{ "operator": ">", "value": "2020-08-26T15:00:00Z"}],"code": [{ "operator": "IN", "value": ["brand", "size", "description","name","picture","color","secondary_color"]}]

Author: it works with and without the space. I updated the list with no space at all between codes.

/api/rest/v1/attributes?search={"updated": [{ "operator": ">", "value": "2020-08-26T15:00:00Z"}],"code": [{ "operator": "IN", "value": ["brand","size","description","name","picture","color","secondary_color"]}]
```  

::: more
[How to filter attributes?](https://api.akeneo.com/documentation/filter.html#filter-attributes)  

> [How to filter families?](https://api.akeneo.com/documentation/filter.html#filter-families)  
> [How to get back attributes data?](https://api.akeneo.com/api-reference.html#get_attributes)

Reviewer: ```suggestion
[How to filter families?](https://api.akeneo.com/documentation/filter.html#filter-families)
```
